lib/srv/db/common/auth.go

```diff
@@ -505,6 +516,56 @@ func (a *dbAuth) Close() error {
 	return a.cfg.Clients.Close()
 }
 
+// generateCloudSQLClientCertificate returns a new client certificate with RSA key generated
+// using Google's GenerateEphemeralCertRequest sqladmin API. Client certificates as required
+// when enabling SSL in Cloud SQL.
+func (a *dbAuth) generateCloudSQLClientCertificate(ctx context.Context, sessionCtx *Session) (tls.Certificate, error) {
+	svc, err := a.cfg.Clients.GetGCPSQLAdminClient(ctx)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+
+	// generate RSA private key, x509 encoded public key,
+	// and append to certificate request
+	pkey, err := rsa.GenerateKey(rand.Reader, constants.RSAKeySize)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	pkix, err := x509.MarshalPKIXPublicKey(pkey.Public())
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	gcp := sessionCtx.Database.GetGCP()
+	req := svc.Connect.GenerateEphemeralCert(gcp.ProjectID, gcp.InstanceID, &sqladmin.GenerateEphemeralCertRequest{
```

Comment:
Does this operation require any additional GCP permissions other than those we have specified in [this](https://goteleport.com/docs/database-access/guides/postgres-cloudsql/#create-service-account-for-teleport-database-service) and [this](https://goteleport.com/docs/database-access/guides/mysql-cloudsql/#create-service-account-for-teleport-database-service) guides? If so, let's update the documentation as well.

Reply:
@jimbishopp I'm also thinking that this will break Cloud SQL access for everyone who's missing permissions to generate ephemeral certs. IIRC we discussed an option where we would look at the instance and determine whether we need to generate the client cert or not depending on whether "enforce SSL connections" setting is enabled or not. I'm leaning towards that would be a better approach than generating it every time.

Reply:
Updated in latest push.